teps/0066-dogfooding-tekton.md

```diff
@@ -0,0 +1,320 @@
+- Describe the design principles for CI/CD services based on Tekton
+- Identify the services implemented through Tekton
+- Identify areas were more work is needed and define a future roadmap
+
+### Non-Goals
+
+This TEP does not aim to design specific services based on Tekton, that work will be tracked
+in dedicated TEPs where needed.
+
+### Use Cases (optional)
+
+Use cases that the dogfooding effort implements are:
+
+- Experience Tekton from the operator, author and user point of view, which allows the
+  Tekton community to:
+  - Identify missing functional features and usability issues
+  - Identify operational problems and security concerns
+  - Discover bugs that can be better identified in a running system, such as upgrade issues,
+    a problems related to maintainability and scale
+- Offer CI/CD services to the Tekton community with minimal dependencies
+  to other products, and thus greater control on the outcome
+- Build a repository of examples that embody best practices for Tekton use to tap in
+
+## Requirements
+
+- Provide reliable CI/CD services to the Tekton community
+- Dogfood as many project projects as possible, from the most stable and mature to
+  the experimental ones
+- Provide a mean to experiment with alpha and experimental features and build a way
+  forward to stable
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Notes/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+
+We want to offer continuous integration and continuous delivery services to
+the Tekton community, which are essential to the development of Tekton itself.
+This goes in contrast with the need of dogfooding alpha stability and even
+experimental services. Mitigations for this are:
+
+- Use a multi-cluster setup:
+  - One cluster runs nightly releases and experimental code. This cluster may
+    offer services, but no essential ones, so that Tekton development may continue
+    normally if this cluster is broken.
+  - One cluster runs major releases of Tekton components (when available).
+    Deployments to this cluster are vetted by humans. Write / admin access to this
```

Comment:
this is kind of a nit but id like to see a world where eventually humans dont have to vet deployments to the cluster - i think it makes sense to not be on the absolute cutting edge, but if we are doing nightly releases to the less stable cluster, it might be reasonable to do auto deployments of major releases to the stable cluster

but that's always something we could add later :D

Reply:
Yeah, I thought a bit about that. 
It's definitely something that would be good to have. 

If we automated heath checks on various services, we could automatically deploy and rollback.
We could run a test version of our services in robocat, and generate test load there, and that would give us confidence in deploying to dogfooding.

I could add this to the goals, but I wanted to focus on goals specific to dogfooding, and being able to achieve this doesn't feel like something that would necessarily be in the domain of Tekton (setting up monitors, handling deployment and rollback).

Maybe we could start a TEP about where and how to improve our devops metrics?

Reply:
Done